tensorflow/python/ops/nn_grad.py

```diff
@@ -491,6 +491,19 @@ def _AvgPoolGrad(op, grad):
       data_format=op.get_attr("data_format"))
 
 
+@ops.RegisterGradient("AvgPoolV2")
+def _AvgPoolGradV2(op, grad):
```

Comment:
We should add tests that actually take in ksizes and strides as input tensors during session running, but not like the current tests where ksizes and strides are still provided when building the graph? So that we can test whether the change works. For example, we could define ksizes and strides as variables and feed values only when we run the session.

This in general holds for all the operators you changed: MaxPool, MaxPoolGrad, MaxPoolGradGrad, AvgPool, AvgPoolGrad, AvgPoolGradGrad.